Footprint chunk test failing

### What happened?

We seem to be getting the following failed test when running the CI at the moment. One of the more recent PRs #1070 did update how the chunking was used when standardising footprint data but this test does pass locally.
```
FAILED tests/standardise/test_standardise.py::test_standardise_footprints_chunk - AssertionError: assert {'height': (2...e': (1, 1, 1)} == {'height': (2...time': (2, 1)}
```

### What did you expect to happen?

_No response_

### Minimal Complete Verifiable Example

_No response_

### Relevant log output

_No response_

### Anything else we need to know?

_No response_

### Environment

<details>



</details>

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -26,6 +26,11 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - The `data_source` keyword is now included as "internal" when using `standardise_surface` to distinguish this from data retrieved from external sources (e.g. "icos", "noaa_obspack"). [PR #1083](https://github.com/openghg/openghg/pull/1083)
 - Added interactive timeseries plots in Search and Plotting tutorial. - [PR #953](https://github.com/openghg/openghg/pull/953) 
 - Pinned the `icoscp` version within requirements to 0.1.17 based on new authentication requirements. [PR #1084](https://github.com/openghg/openghg/pull/1084)
+- The sort flag can now be passed via the SearchResults.retrieve interfaces to choose whether the data is returned sorted along the time axis. [PR #1090](https://github.com/openghg/openghg/pull/1090)
+
+### Fixed
+
+- Bug in test when checking customised chunks were stored correctly in the zarr store. dask v2024.8 now changed the chunk shape after this was sorted was test was updated to ensure this didn't sort when retrieving the data. [PR #1090](https://github.com/openghg/openghg/pull/1090)
 
 ## [0.8.2] - 2024-06-06
 

--- a/openghg/dataobjects/_basedata.py
+++ b/openghg/dataobjects/_basedata.py
@@ -37,7 +37,7 @@ class _BaseData:
             version: Version of data requested from Datasource
             start_date: Start date of data to retrieve
             end_date: End date of data to retrieve
-            sort: Sort the resulting Dataset by the time dimension, defaults to False
+            sort: Sort the resulting Dataset by the time dimension, defaults to True
             elevate_inlet: Force the elevation of the inlet attribute
             attrs_to_check: Attributes to check for duplicates. If duplicates are present
                 a new data variable will be created containing the values from each dataset
@@ -83,6 +83,11 @@ class _BaseData:
             self.data = self._zarrstore.get(version=version)
             if slice_time:
                 # If slicing by time, this must be sorted along the time dimension
+                if sort is False:
+                    logger.warning(
+                        "Ignoring sort={sort} input as it is necessary to sort the data when extracting a start and end date range."
+                    )
+
                 self.data = self.data.sortby("time")
                 sorted = True
 

--- a/openghg/dataobjects/_basedata.py
+++ b/openghg/dataobjects/_basedata.py
@@ -85,7 +85,7 @@ class _BaseData:
                 # If slicing by time, this must be sorted along the time dimension
                 if sort is False:
                     logger.warning(
-                        "Ignoring sort={sort} input as it is necessary to sort the data when extracting a start and end date range."
+                        f"Ignoring sort={sort} input as it is necessary to sort the data when extracting a start and end date range."
                     )
 
                 self.data = self.data.sortby("time")

--- a/openghg/dataobjects/_searchresults.py
+++ b/openghg/dataobjects/_searchresults.py
@@ -110,38 +110,39 @@ class SearchResults:
         self,
         dataframe: Optional[DataFrame] = None,
         version: str = "latest",
+        sort: bool = True,
         **kwargs: Any,
     ) -> Union[ObsData, List[ObsData]]:
         """Retrieve data from object store using a filtered pandas DataFrame
 
         Args:
             dataframe: pandas DataFrame
-            sort: Sort data by date in retrieved Dataset
-            elevate_inlet: Elevate inlet to a variable within the Dataset, useful
-            for ranked data.
+            version: Version of data requested from Datasource. Default = "latest".
+            sort: Sort data by time in retrieved Dataset
+            **kwargs: Metadata values to search for
         Returns:
             ObsData / List[ObsData]: ObsData object(s)
         """
         if dataframe is not None:
             uuids = dataframe["uuid"].to_list()
-            return self._retrieve_by_uuid(uuids=uuids, version=version)
+            return self._retrieve_by_uuid(uuids=uuids, version=version, sort=sort)
         else:
-            return self._retrieve_by_term(version=version, **kwargs)
+            return self._retrieve_by_term(version=version, sort=sort, **kwargs)
 
     def retrieve_all(
         self,
         version: str = "latest",
+        sort: bool = True,
     ) -> Union[ObsData, List[ObsData]]:
         """Retrieves all data found during the search
 
         Args:
+            version: Version of data requested from Datasource. Default = "latest".
             sort: Sort by time. Note that this may be very memory hungry for large Datasets.
-            elevate_inlet: Elevate inlet to a variable within the Dataset, useful
-            for ranked data.
         Returns:
             ObsData / List[ObsData]: ObsData object(s)
         """
-        return self._retrieve_by_uuid(uuids=self.metadata.keys(), version=version)
+        return self._retrieve_by_uuid(uuids=self.metadata.keys(), version=version, sort=sort)
 
     def uuids(self) -> List:
         """Return the UUIDs of the found data
@@ -151,14 +152,18 @@ class SearchResults:
         """
         return list(self.metadata.keys())
 
-    def _retrieve_by_term(self, version: str, **kwargs: Any) -> Union[ObsData, List[ObsData]]:
+    def _retrieve_by_term(
+        self, version: str, sort: bool = True, **kwargs: Any
+    ) -> Union[ObsData, List[ObsData]]:
         """Retrieve data from the object store by search term. This function scans the
         metadata of the retrieved results, retrieves the UUID associated with that data,
         pulls it from the object store, recombines it into an xarray Dataset and returns
         ObsData object(s).
 
         Args:
-            kwargs: Metadata values to search for
+            version: Version of data requested from Datasource. Default = "latest".
+            sort: Sort by time. Note that this may be very memory hungry for large Datasets.
+            **kwargs: Metadata values to search for
         """
         uuids = set()
         # Make sure we don't have any Nones
@@ -187,15 +192,19 @@ class SearchResults:
                 uuids.add(uid)
 
         # Now we can retrieve the data using the UUIDs
-        return self._retrieve_by_uuid(uuids=list(uuids), version=version)
+        return self._retrieve_by_uuid(uuids=list(uuids), version=version, sort=sort)
 
-    def _retrieve_by_uuid(self, uuids: Iterable, version: str) -> Union[ObsData, List[ObsData]]:
+    def _retrieve_by_uuid(
+        self, uuids: Iterable, version: str, sort: bool = True
+    ) -> Union[ObsData, List[ObsData]]:
         """Internal retrieval function that uses the passed in UUIDs to retrieve
         the keys from the key_data dictionary, pull the data from the object store,
         create ObsData object(s) and return the result.
 
         Args:
             uuids: UUIDs of Datasources in the object store
+            version: Version of data requested from Datasource. Default = "latest".
+            sort: Sort by time. Note that this may be very memory hungry for large Datasets.
         Returns:
             ObsData / List[ObsData]: ObsData object(s)
         """
@@ -215,6 +224,7 @@ class SearchResults:
                     metadata=metadata,
                     start_date=self._start_date,
                     end_date=self._end_date,
+                    sort=sort,
                 )
             )
 

--- a/tests/standardise/test_standardise.py
+++ b/tests/standardise/test_standardise.py
@@ -399,7 +399,10 @@ def test_standardise_footprints_chunk(caplog):
     )
 
     search_results = search_footprints(model="UKV-chunked", store="user")
-    fp_data = search_results.retrieve_all()
+
+    # Note: have to pass sort=False here for dask>=2024.8 as this returns different
+    # chunks for time (1, 1, 1) rather than original chunks we're trying to check.
+    fp_data = search_results.retrieve_all(sort=False)
 
     assert dict(fp_data.data.chunks) == {"time": (2, 1), "lat": (12,), "lon": (12,), "height": (20,)}
 

